src/main/java/core/basesyntax/operation/PurchaseHandler.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.service.ShopService;
+import core.basesyntax.transaction.FruitTransaction;
+
+public class PurchaseHandler implements OperationHandler {
+    private ShopService shopService;
+
+    public PurchaseHandler(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int transactionQuantity = transaction.getQuantity();
+        int currentQuantity = shopService.getQuantity(fruit);
+        if (transaction.getOperation() == FruitTransaction.Operation.PURCHASE) {

```

Comment:
The condition `transaction.getOperation() == FruitTransaction.Operation.PURCHASE` is unnecessary. The `PurchaseHandler` should only handle purchase operations. If another operation is passed, it's an error in the code logic, not in the user input. So, you should remove this condition.